scripts/pysa_fuzzer/fuzzer.py

```diff
@@ -0,0 +1,130 @@
+        return f"for {loop_var} in range({start}, {end}):\n{self.indent(body)}"
+
+    def generate_while_loop(self, depth: int, from_var: str) -> str:
+        condition = self.generate_conditional_expression(1)
+        body = self.generate_complex_expression(depth - 1, from_var)
+        return f"while {condition}:\n{self.indent(body)}"
+
+    def generate_try_except(self, depth: int, from_var: str) -> str:
+        try_block = self.generate_complex_expression(depth - 1, from_var)
+        except_block = self.generate_complex_expression(depth - 1, from_var)
+        return f"try:\n{self.indent(try_block)}\nexcept Exception as e:\n{self.indent(except_block)}"
+
+    def generate_function_definition(self, depth: int, from_var: str) -> str:
+        func_name = random.choice(['foo', 'bar', 'baz', 'qux'])
+        params = ', '.join(random.sample(self.variables, random.randint(1, 3)))
+        body = self.generate_complex_expression(depth - 1, from_var)
+        return f"def {func_name}({params}):\n{self.indent(body)}"
+
+    def generate_expression(self, depth: int) -> str:
+        if depth == 0:
+            return random.choice([str(random.randint(0, 10)), random.choice(list(self.defined_variables))])
+        choices = [
+            lambda: self.generate_arithmetic_expression(depth - 1),
+            lambda: self.generate_variable_expression(depth - 1),
+            lambda: self.generate_function_call(depth - 1, ''),
+        ]
+        return random.choice(choices)()
+
+    def generate_statement(self, depth: int, from_var: str) -> str:
+        choices = [
+            lambda: self.generate_assignment(from_var),
+            lambda: self.generate_if_statement(depth, from_var),
+            lambda: self.generate_for_loop(depth, from_var),
+            lambda: self.generate_while_loop(depth, from_var),
+            lambda: self.generate_try_except(depth, from_var),
+            lambda: self.generate_function_definition(depth, from_var),
+        ]
+        return random.choice(choices)()
+
+    def generate_complex_expression(self, depth: int, from_var: str) -> str:
+        if depth == 0:
+            return self.generate_assignment(from_var)
+        parts: List[Callable[[], str]] = [
+            lambda: self.generate_assignment(from_var),
+            lambda: self.generate_if_statement(depth, from_var),
+            lambda: self.generate_function_call(depth, from_var),
+            lambda: self.generate_for_loop(depth, from_var),
+            lambda: self.generate_while_loop(depth, from_var),
+            lambda: self.generate_try_except(depth, from_var),
+            lambda: self.generate_function_definition(depth, from_var)
+        ]
+        random.shuffle(parts)
+        return '\n'.join([part() for part in parts[:random.randint(2, 3)]])
+
+    def generate_source_to_sink(self) -> str:
+        self.current_var_index = 0  # Reset the index for each new generation
+        self.defined_variables.clear()

```

Comment:
That's not really necessary. We should just instantiate a new `CodeGenerator` each time, so the state is properly initialized.